x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -113,22 +131,20 @@ public boolean acquire(final EvictionListener listener) throws IOException {
         ensureOpen();
         boolean success = false;
         if (refCounter.tryIncRef()) {
```

Comment:
This seems unchanged, but I find it slightly odd to have both the option of returning false from this method as well as throwing `AlreadyClosedException` when this `CacheFile` is being evicted. Am I missing something here?

Reply:
Yea I'm not sure about this either. I think we should just always throw here, since the only way we could conceivably return `false` is if we evict between `ensureOpen()` and trying to increment the ref counter (which is close to impossible).
Maybe change that in a follow-up though, could simplify this and also the `release` method which always returns `true` currently as well and can be simplified to a `void`.

Reply:
Yes, I think this is a left over when the IndexInput tried to re-acquire a cache file if it failed to add itself as a listener. I guess we were just returning `false` at that time in case of eviction.

Anyway, I'm +1 on throwing.